src/cpp/core/system/PosixFileScanner.cpp

```diff
@@ -124,6 +124,10 @@ Error scanFiles(const tree<FileInfo>::iterator_base& fromNode,
    // iterate over the names
    for (const std::string& name : names)
    {
+      // check for interrupt
+      if (boost::this_thread::interruption_requested())
+         return core::systemError(boost::system::errc::interrupted, ERROR_LOCATION);
```

Comment:
What are the expected consequences of this change for the non-Windows cases? Given this delay issue was Windows-specific, would it be safer to leave this part as-is and leave this change (potentially) for 1.4/master?

Thinking specifically of the QA test matrix; if code changes are definitively Windows-only, then (presumably) we can tell them to worry specifically only about Windows desktop with this change, but modifying this here seems to bring testing of Mac and Linux desktops into scope?

Reply:
I think ultimately other platforms run risking the same issue: the file monitor thread isn't able to clean up in time (because it's stuck in the file scanner), but we try to exit anyways and run into problems trying to shut down while the thread is still active.

That said, given that this is just theoretical and (AFAIK) we haven't observed this issue in v1.3 thus far, we could back this part out.